web/src/main/java/org/fao/geonet/monitor/onlineresource/OnlineResourceCheckerWps.java

```diff
@@ -0,0 +1,179 @@
+                }
+            }
+
+            logger.debug(String.format("%s -> %d", url, connection.getResponseCode()));
+
+            if (connection.getResponseCode() != 200) {
+                logger.info(String.format("link broken uuid='%s', url='%s', error='bad response code %d'",
+                        this.uuid, this.url, connection.getResponseCode()));
+                return false;
+            }
+
+            // Checking if file is readable
+            if (connection.getContentType().equals("text/csv")) {
+                InputStream is = null;
+                CSVReader reader;
+                try {
+                    is = connection.getInputStream();
+                    reader = new CSVReader(new InputStreamReader(connection.getInputStream()));
+                    while ((reader.readNext()) != null) {
+                        break;
+                    }
+                } finally {
+                    if (is != null) {
+                        is.close();
+                    }
+                }
+                return true;
+            } else {
+                logger.info(String.format("link broken uuid='%s', url='%s', error='unexpected content-type'", this.uuid, this.url));
+                return false;
+            }
+
+        } catch (Exception e) {
+            logger.info(String.format("link broken uuid='%s', url='%s', error='%s' stack='%s'",
+                    uuid, url, e.getMessage(), OnlineResourceCheckerUtils.exceptionToString(e)));
+        }
+
+        return false;
+    }
+
+    private String getRequestXml(String time) {
+        String literalTime = time + "00Z";
+        return "<?xml version=\"1.0\" encoding=\"UTF-8\"?><wps:Execute version=\"1.0.0\" service=\"WPS\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xmlns=\"http://www.opengis.net/wps/1.0.0\" xmlns:wfs=\"http://www.opengis.net/wfs\" xmlns:wps=\"http://www.opengis.net/wps/1.0.0\" xmlns:ows=\"http://www.opengis.net/ows/1.1\" xmlns:gml=\"http://www.opengis.net/gml\" xmlns:ogc=\"http://www.opengis.net/ogc\" xmlns:wcs=\"http://www.opengis.net/wcs/1.1.1\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" xsi:schemaLocation=\"http://www.opengis.net/wps/1.0.0 http://schemas.opengis.net/wps/1.0.0/wpsAll.xsd\">\n" +
+                "  <ows:Identifier>gs:GoGoDuck</ows:Identifier>\n" +
+                "  <wps:DataInputs>\n" +
+                "    <wps:Input>\n" +
+                "      <ows:Identifier>layer</ows:Identifier>\n" +
+                "      <wps:Data>\n" +
+                "        <wps:LiteralData>" + layer + "</wps:LiteralData>\n" +
+                "      </wps:Data>\n" +
+                "    </wps:Input>\n" +
+                "    <wps:Input>\n" +
+                "      <ows:Identifier>subset</ows:Identifier>\n" +
+                "      <wps:Data>\n" +
+                "        <wps:LiteralData>TIME," + literalTime + "," + literalTime + ";LATITUDE,-31.6855,-31.6855;LONGITUDE,114.8291,114.8291</wps:LiteralData>\n" +
```

Comment:
Are we sure that there will always be data for gogoduck requests in this spatial extent?